flow/Platform.cpp

```diff
@@ -2447,8 +2446,8 @@ std::vector<std::string> listFiles( std::string const& directory, std::string co
 	return findFiles( directory, extension, &acceptFile );
 }
 
-std::vector<std::string> listDirectories( std::string const& directory ) {
-	return findFiles( directory, "", &acceptDirectory );
+std::vector<std::string> listDirectories(std::string const& directory, std::string const& extension) {
```

Comment:
It seems that `if ((*accept_file)(buf.st_mode, name, extension))`  didn't use the extension, where `accept_file` is defined here: https://github.com/ajbeamon/foundationdb/blob/f4655783a4f235b796e0d160b91bc88cc325ab28/flow/Platform.cpp#L2370-L2379 

How is extension used to filter out the folders that are not valid (rocked data) folders?

Reply:
Good catch. Thanks! It should be fixed now.